### PR TITLE
Team screen redesign

### PR DESCRIPTION
**Pull request by ice-minos**
Redesign the team screen exactly as in Figma

### CODE REVIEW

File: src/components/ListItems/UserListItem/index.tsx [hunk 1]
```diff
@@ -16,19 +17,23 @@ import {rem} from 'rn-units';
...
     userId: string;
     name?: string | null | ReactNode;
-    note?: string | null | ReactNode;
+    phoneNumber?: string | null | ReactNode;
     profilePictureUrl?: string | null;
     active?: boolean;
     AdditionalInfoComponent?: ReactNode;
+    country?: string | null;
   }) => {
+    const countryFlag = countries.find(item => item.isoCode === country)?.flag;
```
Comment (ice-ares) [inline src/components/ListItems/UserListItem/index.tsx:35]: compare them case insensitively

File: src/screens/AuthFlow/ConfirmPhone/hooks/useConfirmPhone.ts [hunk 2]
```diff
@@ -47,15 +50,29 @@ export const useConfirmPhone = () => {
   const onSetCode = (value: string) => {
     setCode(value);
     if (validateError) {
-      dispatch(AccountActions.SIGN_IN_PHONE.CLEAR_ERROR.create());
+      if (workMode === 'TEAM') {
```
[the remark this answers is not in the document]
Comment (ice-ares) [reply src/screens/AuthFlow/ConfirmPhone/hooks/useConfirmPhone.ts:53]: also, if u want to make this a generic hook, dont add any specific logic and move it into a generic location

File: src/screens/Team/components/Header/components/InfoItem/index.tsx [hunk 3]
```diff
@@ -46,9 +45,11 @@ export function InfoItem({type}: InfoItemProps): React.ReactElement {
         return <Text style={styles.referrals}>{refsCount}</Text>;
       case InfoItemType.earnings:
         return (
-          <Text style={styles.earnings}>
-            {earningsValue} <IceLabel iconSize={16} />
-          </Text>
+          <View style={styles.indexContainer}>
+            <Text style={styles.earnings}>{earningsValue}</Text>
+            <Text style={styles.indexText}>42</Text>
+            <Text style={styles.iceText}> ice</Text>
```
Comment (ice-ares) [inline src/screens/Team/components/Header/components/InfoItem/index.tsx:51]: no, ice is never just text, use the special component

File: src/screens/Team/components/TierList/index.tsx [hunk 4]
```diff
@@ -63,18 +63,26 @@ export const TierList = memo(
...
-          profilePictureUrl={item.profilePictureUrl}
-          active={item.active}
-          AdditionalInfoComponent={<UserListPingButton pinged={item.pinged} />}
-        />
-      );
-    }, []);
+    const renderItem = useCallback(
+      ({item}: {item: User}) => {
+        return (
+          <UserListItem
```
Comment (ice-ares) [inline src/screens/Team/components/TierList/index.tsx:69]: refactor the component to accept the whole user object

File: src/screens/AuthFlow/ConfirmPhone/hooks/useConfirmPhone.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (ice-minos) [reply src/screens/AuthFlow/ConfirmPhone/hooks/useConfirmPhone.ts:53]: maybe I should create a separate hook for this task?

[the remark this answers is not in the document]
Comment (ice-ares) [reply src/screens/AuthFlow/ConfirmPhone/hooks/useConfirmPhone.ts:53]: just move this to a generic place, and make it generic as well

File: src/components/ListItems/UserListItem/index.tsx (hunk 1 above)
Comment (ice-minos) [reply src/components/ListItems/UserListItem/index.tsx]: done

File: src/screens/Team/components/Header/components/InfoItem/index.tsx (hunk 3 above)
Comment (ice-minos) [reply src/screens/Team/components/Header/components/InfoItem/index.tsx]: done

File: src/screens/Team/components/Header/components/InfoItem/index.tsx [hunk 5]
```diff
@@ -19,8 +18,8 @@ type InfoItemProps = {
 };
 
 export function InfoItem({type}: InfoItemProps): React.ReactElement {
-  const refsCount = 0;
-  const earningsValue = 121985;
+  const refsCount = '139,205';
```
[the remark this answers is not in the document]
Comment (ice-minos) [reply src/screens/Team/components/Header/components/InfoItem/index.tsx]: done

File: src/screens/Team/components/TierList/index.tsx (hunk 4 above)
Comment (ice-minos) [reply src/screens/Team/components/TierList/index.tsx:69]: done

File: src/screens/Team/components/SearchResults/components/SearchUserItem.tsx [hunk 6]
```diff
@@ -3,43 +3,17 @@
...
-          <Text style={styles.referralTypeText}>{' ' + referralType}</Text>
-        </>
-      }
-      note={user.phoneNumber}
-      profilePictureUrl={user.profilePictureUrl}
-      active={user.active}
-      userId={user.id}
-      key={user.id}
+      user={user}
       AdditionalInfoComponent={<UserListPingButton pinged={user.pinged} />}
```
Comment (ice-ares) [inline src/screens/Team/components/SearchResults/components/SearchUserItem.tsx:16]: render this only if user.pinged is true or false. This is the same for all places where u use user.pinged

File: src/screens/HomeFlow/Home/components/Header/index.tsx [hunk 7]
```diff
@@ -48,11 +48,14 @@ export const HomeHeader = memo(({translateY, transitionOffset}: Props) => {
     <View style={[styles.container, topOffset.current]}>
       <View style={styles.body}>
         {user?.profilePictureUrl && (
-          <Avatar
-            uri={user.profilePictureUrl}
-            size={rem(36)}
-            borderRadius={rem(16)}
-          />
+          <Touchable onPress={onGreetingPress}>
```
Comment (ice-orion) [inline src/screens/HomeFlow/Home/components/Header/index.tsx:51]: `onGreetingPress` when we press on `Avatar`? May be it makes sense to change the name of the function

File: src/screens/Team/assets/svg/TeamHeaderReferralsIcon.tsx [hunk 8]
```diff
@@ -4,10 +4,16 @@ import * as React from 'react';
 import Svg, {Path, SvgProps} from 'react-native-svg';
 
 export const TeamHeaderReferralsIcon = (props: SvgProps) => (
```
Comment (ice-orion) [inline src/screens/Team/assets/svg/TeamHeaderReferralsIcon.tsx:6]: image

Is it this icon? If so, we already have it and using on the tabbar -
image

File: src/screens/Team/assets/svg/TeamHeaderEarningsIcon.tsx [hunk 9]
```diff
@@ -1,20 +1,13 @@
 // SPDX-License-Identifier: BUSL-1.1
 
 import * as React from 'react';
-import Svg, {ClipPath, Defs, G, Path, SvgProps} from 'react-native-svg';
+import Svg, {Path, SvgProps} from 'react-native-svg';
 
 export const TeamHeaderEarningsIcon = (props: SvgProps) => (
```
Comment (ice-orion) [inline src/screens/Team/assets/svg/TeamHeaderEarningsIcon.tsx:6]: Is it this icon?- 
image

if so, we already have the same icon on balance popup - 
image

File: src/screens/Team/components/Contacts/components/ContactsList/index.tsx [hunk 10]
```diff
@@ -34,9 +33,10 @@ import {rem} from 'rn-units';
 
 type Props = {
   focused: boolean;
+  onScroll: CallableFunction;
```
Comment (ice-orion) [inline src/screens/Team/components/Contacts/components/ContactsList/index.tsx:36]: don't use `CallableFunction`, write function signatures

File: src/screens/Team/components/Contacts/components/ContactsList/index.tsx [hunk 11]
```diff
@@ -80,13 +80,11 @@ export const ContactsList = ({focused}: Props) => {
...
           <UserListItem
-            userId={item.id}
-            name={item.username}
-            note={item.active ? t('users.active') : t('users.inactive')}
-            profilePictureUrl={item.profilePictureUrl}
-            active={item.active}
+            user={item}
             AdditionalInfoComponent={
-              <UserListPingButton pinged={item.pinged} />
+              item.pinged === true || item.pinged === false ? (
```
Comment (ice-orion) [inline src/screens/Team/components/Contacts/components/ContactsList/index.tsx:85]: `item.pinged != null && <UserListPingButton...`

File: src/screens/Team/components/Contacts/components/ContactsList/index.tsx [hunk 12]
```diff
@@ -109,6 +107,7 @@ export const ContactsList = ({focused}: Props) => {
         onEndReached={loadNext}
         onRefresh={refresh}
         refreshing={refreshing}
+        onScroll={event => onScroll(event)}
```
Comment (ice-orion) [inline src/screens/Team/components/Contacts/components/ContactsList/index.tsx:110]: no need to pass `onScroll` to all the components, use one main ScrollView on the `src/screens/Team/index.tsx` and handle scroll there (just like on the Home screen)

File: src/screens/Team/components/Header/components/InfoItem/index.tsx [hunk 13]
```diff
@@ -46,9 +48,11 @@ export function InfoItem({type}: InfoItemProps): React.ReactElement {
         return <Text style={styles.referrals}>{refsCount}</Text>;
       case InfoItemType.earnings:
         return (
-          <Text style={styles.earnings}>
-            {earningsValue} <IceLabel iconSize={16} />
-          </Text>
+          <View style={styles.indexContainer}>
+            <Text style={styles.earnings}>{earningsValue}</Text>
+            <Text style={styles.indexText}>42</Text>
```
Comment (ice-orion) [inline src/screens/Team/components/Header/components/InfoItem/index.tsx:53]: use `FormattedNumber` component instead

File: src/screens/Team/components/Header/index.tsx [hunk 14]
```diff
@@ -22,58 +23,72 @@ import {useDispatch, useSelector} from 'react-redux';
 type Props = {
   isSearchActive: boolean;
   setSearchActive: (active: boolean) => void;
+  isScrolled: boolean;
 };
 
-export const Header = memo(({isSearchActive, setSearchActive}: Props) => {
-  const dispatch = useDispatch();
+export const Header = memo(
+  ({isSearchActive, setSearchActive, isScrolled}: Props) => {
```
Comment (ice-orion) [inline src/screens/Team/components/Header/index.tsx:30]: Did you discuss the scroll animation with Andrey? I think we shouldn't just hide the header on scrool, but it should work smooth like on the home screen. Moreover it seems that it doesn't work:

https://user-images.githubusercontent.com/102020833/203292188-62a63f5b-a36e-49e3-8bd7-bb564ff9a9ec.mp4

File: src/screens/Team/components/SearchResults/components/SearchUserItem.tsx [hunk 15]
```diff
@@ -3,43 +3,17 @@
...
-import {getCountryByCode} from '@utils/country';
-import {font} from '@utils/styles';
 import React from 'react';
-import {StyleSheet, Text} from 'react-native';
 
 type Props = {
   user: User;
 };
 
 export const SearchUserItem = ({user}: Props) => {
```
Comment (ice-orion) [inline src/screens/Team/components/SearchResults/components/SearchUserItem.tsx:12]: This is almost the duplicate of `UserListItem` now, let's get rid of it and use `UserListItem` directly

File: src/screens/Team/components/SegmentedContent/segments.tsx [hunk 16]
```diff
@@ -6,14 +6,26 @@ import {TierOneIcon} from '@svg/TierOneIcon';
 import {TierTwoIcon} from '@svg/TierTwoIcon';
 import {t} from '@translations/i18n';
 import React from 'react';
-import {Text} from 'react-native';
+import {StyleSheet, Text} from 'react-native';
+import {rem} from 'rn-units';
+
+const styles = StyleSheet.create({
+  text: {
+    fontSize: rem(17),
```
Comment (ice-orion) [inline src/screens/Team/components/SegmentedContent/segments.tsx:14]: use `font` util, did the font size changed for all the segments? probably the font should be also changed in the main SegmentedControl component

File: src/screens/Team/index.tsx [hunk 17]
```diff
@@ -48,4 +69,8 @@ const styles = StyleSheet.create({
     backgroundColor: COLORS.primaryLight,
     flex: 1,
   },
+  cardContainer: {
```
Comment (ice-orion) [inline src/screens/Team/index.tsx:72]: What's the style is this?

File: src/screens/Team/index.tsx [hunk 18]
```diff
@@ -1,36 +1,57 @@
...
 import {isIOS} from 'rn-units';
 
 export const Team = memo(() => {
   useFocusStatusBar({style: 'light-content'});
   const [isCountryCodeSearchVisible, setCountryCodeSearchVisibility] =
     useState<boolean>(false);
   const [isSearchActive, setSearchActive] = useState(false);
+  const [isHeaderCollapse, setIsHeaderCollapse] = useState(false);
+
+  const headerCollapseHandler = useCallback(
```
Comment (ice-orion) [inline src/screens/Team/index.tsx:24]: As i described above this works bad and probably needs to be reconsidered

File: src/store/modules/UtilityProcessStatuses/selectors/index.ts [hunk 19]
```diff
@@ -81,3 +81,12 @@ export const failedReasonSelector = (
 
   return undefined;
 };
+
+export const actionPayloadSelectorErrorMessage = (
```
Comment (ice-orion) [inline src/store/modules/UtilityProcessStatuses/selectors/index.ts:85]: isn't it the same as `failedReasonSelector`?

File: src/store/modules/Validation/actions/index.ts [hunk 20]
```diff
@@ -26,6 +26,8 @@ const PHONE_VALIDATION = createAction('PHONE_VALIDATION', {
     errorMessage,
     errorCode,
   }),
+  CLEAR_ERROR: true,
```
Comment (ice-orion) [inline src/store/modules/Validation/actions/index.ts:29]: Where do these actions are used?

File: src/components/ConfirmPhoneNumber/index.tsx [hunk 21]
```diff
@@ -52,6 +58,10 @@ export const ConfirmPhoneNumber = memo(({onSubmitPress}: Props) => {
   );
 
   useEffect(() => {
+    if (onHideHeader) {
+      onHideHeader(null, true);
```
Comment (ice-hector) [inline src/components/ConfirmPhoneNumber/index.tsx:62]: Let's maybe rename it, and why the first parameter is null? Can we avoid this?

File: src/components/ListItems/UserListItem/index.tsx [hunk 22]
```diff
@@ -113,6 +121,16 @@ const styles = StyleSheet.create({
...
   },
+  outerContainer: {
+    paddingHorizontal: rem(20),
+    paddingVertical: rem(8),
+  },
+  flagContainer: {
+    marginLeft: 12,
+  },
+  flagStile: {
+    fontSize: 26,
```
Comment (ice-hector) [inline src/components/ListItems/UserListItem/index.tsx:132]: Apply font wrapper here

File: src/screens/Team/components/Contacts/index.tsx [hunk 23]
```diff
@@ -27,12 +27,14 @@ type ContactsProps = {
   showCountriesList: (t: boolean) => void;
   isCountriesVisible: boolean;
   focused: boolean;
+  onScroll: CallableFunction;
```
Comment (ice-hector) [inline src/screens/Team/components/Contacts/index.tsx:30]: Avoid using CallableFunction

File: src/screens/Team/components/SegmentedContent/index.tsx [hunk 24]
```diff
@@ -17,6 +17,7 @@ import {SEGMENTS} from './segments';
 type TiersProps = {
   showCountriesList: (t: boolean) => void;
   isCountriesVisible: boolean;
+  onScroll: CallableFunction;
```
Comment (ice-hector) [inline src/screens/Team/components/SegmentedContent/index.tsx:20]: Avoid using CallableFunction

File: src/screens/HomeFlow/Home/components/Header/index.tsx (hunk 7 above)
Comment (ice-orion) [reply src/screens/HomeFlow/Home/components/Header/index.tsx:51]: this is totally optional, but i would name it something like `<Touchable onPress={openProfile}>`

File: src/screens/Team/components/Header/index.tsx [hunk 25]
```diff
@@ -22,58 +23,72 @@ import {useDispatch, useSelector} from 'react-redux';
...
+    useEffect(() => {
+      searchShared.value = withTiming(isSearchActive ? 1 : 0);
+      scrollShared.value = withTiming(isScrolled ? 1 : 0);
+    }, [isScrolled, isSearchActive, scrollShared, searchShared]);
 
-  const onFocus = () => {
-    setSearchActive(true);
-  };
+    const onFocus = () => {
+      setSearchActive(true);
```
Comment (ice-hector) [inline src/screens/Team/components/Header/index.tsx:65]: setIsSearchActive

File: src/screens/Team/components/SearchResults/index.tsx [hunk 26]
```diff
@@ -30,7 +33,17 @@ export const SearchResults = memo(() => {
   });
 
   const renderItem = useCallback(({item}: {item: User}) => {
-    return <SearchUserItem user={item} key={item.id} />;
+    return (
+      <UserListItem
+        user={item}
+        AdditionalInfoComponent={
+          item.pinged !== null ? (
```
Comment (ice-ares) [inline src/screens/Team/components/SearchResults/index.tsx:40]: or undefined. Make sure u add that condition to all places where u use user.pinged

Comment (ice-orion) [reply src/screens/Team/components/SearchResults/index.tsx]: done

File: src/screens/Team/components/TierList/index.tsx [hunk 27]
```diff
@@ -63,18 +55,20 @@ export const TierList = memo(
...
+            user={item}
+            AdditionalInfoComponent={
+              referralType === 'T1' &&
```
Comment (ice-ares) [inline src/screens/Team/components/TierList/index.tsx:64]: remove this check

File: src/components/ListItems/UserListItem/index.tsx [hunk 28]
```diff
@@ -72,7 +73,9 @@ export const UserListItemSkeleton = ({
   containerStyle,
 }: {containerStyle?: StyleProp<ViewStyle>} = {}) => (
   <SkeletonPlaceholder>
-    <View style={[styles.skeleton, containerStyle]} />
+    <View style={[containerStyle]}>
```
Comment (ice-hector) [inline src/components/ListItems/UserListItem/index.tsx:76]: Remove array brackets

File: src/screens/Team/components/TierList/index.tsx (hunk 27 above)
Comment (ice-orion) [reply src/screens/Team/components/TierList/index.tsx]: done

File: src/components/ListItems/UserListItem/index.tsx (hunk 28 above)
Comment (ice-orion) [reply src/components/ListItems/UserListItem/index.tsx]: done

**Review by ice-ares — DISMISSED** [review]

**Review by ice-hector — APPROVED** [review]

**Review by ice-ares — APPROVED** [review]

**Review by ice-orion — APPROVED** [review]

**Pull request merged**